nevergrad/optimization/optimizerlib.py

```diff
@@ -256,6 +257,82 @@ def __init__(
 CMA = ParametrizedCMA().set_name("CMA", register=True)
 DiagonalCMA = ParametrizedCMA(diagonal=True).set_name("DiagonalCMA", register=True)
 
+class _FCMA(base.Optimizer):
```

Comment:
Since most of the code is similar to CMA the code should be shared as much as possible. Avoid code duplication makes it way easier to iterate on the long run. Two ways of doing it here:
- add an argument in the original CMA (`fcmaes: bool = False`), if some arguments are not supported in one of the other version (popsize? can it be added to the original CMA? and diagonal for FCMA?), raise an error/warning if it is set and shouldnt.
- inherit  the original CMA and override methods that are different. I try to  If I'm correct that's only the `es` property.

I have a strong preference for the first option.

Reply:
Ok, will implement the first option.